packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -641,8 +640,15 @@ abstract class ClusterBase extends Resource implements ICluster {
    * @param chart the cdk8s chart.
    * @returns a `KubernetesManifest` construct representing the chart.
    */
-  public addCdk8sChart(id: string, chart: cdk8s.Chart): KubernetesManifest {
-    return this.addManifest(id, ...chart.toJson());
+  public addCdk8sChart(id: string, chart: Construct): KubernetesManifest {
+
+    const cdk8sChart = chart as any;
+
+    if (typeof cdk8sChart.toJson !== 'function') {
```

Comment:
I didn't add a naive constructor validation since it will fail for custom charts that extend `cdk8s.Chart`. A more complex validation might be worth while but we can add that later on.

For now I think this will suffice.